### PR TITLE
ISPN-10953 CLI index command

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-10953

### CODE REVIEW

File: server/tests/src/test/java/org/infinispan/server/cli/CliIT.java [hunk 1]
```diff
@@ -108,7 +108,15 @@ public void testCliInteractive() {
...
          terminal.send("query \"from org.infinispan.rest.search.entity.Person p where p.gender = 'MALE'\"");
          terminal.assertContains("\"total_results\":3,");
+         terminal.clear();
 
+         terminal.send("index stats qcache");
+         terminal.assertContains("\"slowest\" : \"from org.infinispan.rest.search.entity.Person p where p.gender = 'MALE'\"");
+         terminal.clear();
+         terminal.send("index clear qcache");
+         terminal.send("index reindex qcache");
+         terminal.send("index clear-stats qcache");
```
Comment (fax4ever) [inline server/tests/src/test/java/org/infinispan/server/cli/CliIT.java:118]: Maybe we can test that after the clear there are no more match and after the reindex we will have some matches again.

File: cli/src/main/resources/help/index.adoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
+INDEX(1)
```
Comment (fax4ever) [inline cli/src/main/resources/help/index.adoc:1]: Probably I'm missing something. Where is referenced this file?

**Review by fax4ever — COMMENTED** [review]
LGTM, just 2 comments

File: server/tests/src/test/java/org/infinispan/server/cli/CliIT.java (hunk 1 above)
Comment (tristantarrant) [reply server/tests/src/test/java/org/infinispan/server/cli/CliIT.java:118]: unfortunately those are async calls which don't wait for completion. We could introduce an artificial delay, or probably poll the stats to check completion of the task.

File: cli/src/main/resources/help/index.adoc (hunk 2 above)
Comment (tristantarrant) [reply cli/src/main/resources/help/index.adoc:1]: You need to do `help index`

File: cli/src/main/java/org/infinispan/cli/commands/rest/Index.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import org.infinispan.cli.completers.CacheCompleter;
+import org.infinispan.cli.impl.ContextAwareCommandInvocation;
+import org.infinispan.cli.resources.Resource;
+import org.infinispan.client.rest.RestClient;
+import org.infinispan.client.rest.RestResponse;
+import org.kohsuke.MetaInfServices;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 11.0
```
Comment (pruivo) [inline cli/src/main/java/org/infinispan/cli/commands/rest/Index.java:22]: nit: 
```suggestion
 * @since 14.0
```

**Review by pruivo — COMMENTED** [review]
LGTM, just a small nit :)

File: cli/src/main/java/org/infinispan/cli/commands/rest/Index.java (hunk 3 above)
Comment (tristantarrant) [reply cli/src/main/java/org/infinispan/cli/commands/rest/Index.java]: Fixed

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
merged! thanks @tristantarrant !

**Pull request merged**